src/sql/workbench/contrib/notebook/browser/calloutDialog/imageCalloutDialog.ts

```diff
@@ -196,10 +199,12 @@ export class ImageCalloutDialog extends Modal {
 
 	public insert(): void {
 		this.hide('ok');
+		let imgPath = this._imageUrlInputBox.value;
+		let imageName = path.basename(imgPath);
 		this._selectionComplete.resolve({
-			insertEscapedMarkdown: `![](${escapeUrl(this._imageUrlInputBox.value)})`,
-			imagePath: this._imageUrlInputBox.value,
-			embedImage: this._imageEmbedCheckbox.checked
+			embedImage: this._imageEmbedCheckbox.checked,
+			insertEscapedMarkdown: this._imageEmbedCheckbox.checked ? `![${imageName}](attachment:${imageName.replace(' ', '')})` : `![](${imgPath.replace(' ', '&#32;')})`,
```

Comment:
This stuff could use some comments about why you're doing these kind of replacements and what the end result is going to be